src/miner_util.erl

```diff
@@ -104,16 +104,30 @@ metadata_fun() ->
         Map = blockchain_worker:signed_metadata_fun(),
         case application:get_env(miner, mode, gateway) of
             validator ->
+                RPC =
+                    case lists:filter(fun libp2p_transport_tcp:is_public/1,
+                                      libp2p_swarm:listen_addrs(blockchain_swarm)) of
+                        [] -> #{};
+                        [First | _] ->
```

Comment:
does this filter every realistically return more than one public address? how does a containerized validator typically prevent the container addresses from being tagged as a public listening address and broadcasting it?

Reply:
in the past it has done so enough that I want to handle it explictly, it's a bug, it might have been fixed, but some of this code is gnarly and problematic.